### PR TITLE
docs(kic): update prometheus labels for KIC 2.9.x

### PR DESCRIPTION
**Pull request by pmalek**
### Description

Update prometheus labels for KIC 2.9.x to account for https://github.com/Kong/kubernetes-ingress-controller/pull/3521

This also prunes some trailing whitespace in the changed doc.

Closes https://github.com/Kong/kubernetes-ingress-controller/issues/3417

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

**Review by czeslavo — APPROVED** [review]

**Comment by pmalek** [comment]
@acgoldsmith Can we get this merged? The conditional formatting allows us to merge this before KIC's 2.9 release

File: app/_src/kubernetes-ingress-controller/references/prometheus.md [hunk 1]
```diff
@@ -17,29 +17,35 @@ This document is a reference for the former type.
 
 This metric provides the following labels:
 
-* `protocol` describes the configuration protocol in use, which can be `db-less` or `deck`. 
+* `protocol` describes the configuration protocol in use, which can be `db-less` or `deck`.
+{% if_version gte:2.9.x %}
+* `dataplane` describes the dataplane that was the target of configuration push.
```
Comment (lena-larionova) [inline app/_src/kubernetes-ingress-controller/references/prometheus.md:22]:
```suggestion
* `dataplane` describes the data plane that was the target of configuration push.
```
Data plane, two words, is our standard term (though you'll find a lot of instances of both dataplane and data plane in the docs, from before we standardized on just one).

Mesh/Kuma is the exception, where they specifically use "dataplane" to refer to the entity.

File: app/_src/kubernetes-ingress-controller/references/prometheus.md [hunk 2]
```diff
@@ -17,29 +17,35 @@ This document is a reference for the former type.
...
+{% endif_version %}
 * `success` logs the status of configuration updates. If `success` is `false`, an unrecoverable error occurred.  If `success` is `true`, the push succeeded with no errors.
 {% if_version gte:2.7.x %}
-* `failure_reason` is populated in case `success="false"`. It describes the reason for the failure: 
-    * `conflict`: A configuration conflict that must be manually fixed. 
+* `failure_reason` is populated in case `success="false"`. It describes the reason for the failure:
```
Comment (lena-larionova) [inline app/_src/kubernetes-ingress-controller/references/prometheus.md:26]:
```suggestion
* `failure_reason` is populated if `success="false"`. It describes the reason for the failure:
```

File: app/_src/kubernetes-ingress-controller/references/prometheus.md [hunk 3]
```diff
@@ -17,29 +17,35 @@ This document is a reference for the former type.
...
+    * `conflict`: A configuration conflict that must be manually fixed.
     * `network`: A network related issues, such as {{site.base_gateway}} is offline.
-    * `other`: Other issues, such as {{site.base_gateway}} reports a non-conflict error. 
+    * `other`: Other issues, such as {{site.base_gateway}} reports a non-conflict error.
```
Comment (lena-larionova) [inline app/_src/kubernetes-ingress-controller/references/prometheus.md:29]:
```suggestion
    * `other`: Other issues, such as {{site.base_gateway}} reporting a non-conflict error.
```

File: app/_src/kubernetes-ingress-controller/references/prometheus.md [hunk 4]
```diff
@@ -17,29 +17,35 @@ This document is a reference for the former type.
...
-`ingress_controller_configuration_push_duration_milliseconds` (type: `histogram`) is the amount of time, in milliseconds, that it takes to push the configuration to {{site.base_gateway}}. 
+`ingress_controller_configuration_push_duration_milliseconds` (type: `histogram`) is the amount of time, in milliseconds, that it takes to push the configuration to {{site.base_gateway}}.
 
 This metric provides the following labels:
- 
-* `success` logs the status of configuration updates. If `success` is `false`, an unrecoverable error occurred.  If `success` is `true`, the push succeeded with no errors.
+
 * `protocol` describes the configuration protocol in use, which can be `db-less` or `deck`.
+{% if_version gte:2.9.x %}
+* `dataplane` describes the dataplane that was the target of configuration push.
```
Comment (lena-larionova) [inline app/_src/kubernetes-ingress-controller/references/prometheus.md:46]:
```suggestion
* `dataplane` describes the data plane that was the target of configuration push.
```

**Review by lena-larionova — APPROVED** [review]
Some minor grammar edits, otherwise LGTM. Merging them in directly since they don't change meaning.

**Pull request merged**